src/main/java/uk/gov/register/store/postgres/PostgresDriverNonTransactional.java

```diff
@@ -57,6 +57,11 @@ protected void useHandle(HandleConsumer callback) {
     }
 
     @Override
+    protected <ReturnType> ReturnType withHandleRead(HandleCallback<ReturnType> callback) {
+        return dbi.withHandle(callback);
+    }
```

Comment:
Good spot on this bug! It certainly wasn't meant to be going to the database for every call to `getItemBySha256()`.

I'm not sure about adding a `withHandleRead()` though, since it doesn't make much sense for `PostgresDriverNonTransactional`.

Could we instead add something like a `committedEntryCount` property to the `PostgresDriverTransactional` which we can initialise to `super.getTotalEntries()` in the constructor. Then for `PostgresDriverTransactional.getTotalEntries()` we could return `committedEntryCount + stagedItems.size()`. We'd also have to reset `committedEntryCount` on the `writeEntries()` method too. What do you think?